SwiftyTesseract/SwiftyTesseract/SwiftyTesseract.swift

```diff
@@ -186,7 +190,64 @@ public class SwiftyTesseract {
     completionHandler(swiftString)
     
   }
-
+  
+  /// Takes an array UIImages and returns the PDF as a data object
+  ///
+  /// - Parameter images: Array of UIImages that should be analyzed
+  /// - Returns: PDF as Data object
+  /// - Throws: SwiftyTesseractError in case something went wrong
+  public func createPDF(from images: [UIImage]) throws -> Data {
+    let _ = semaphore.wait(timeout: .distantFuture)
+    
+    // create unique file path
+    let file = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(NSUUID().uuidString)
+    
+    guard let renderer = TessPDFRendererCreate(file.path, bundle.pathToTrainedData, 0) else {
+        throw SwiftyTesseractError.unableToCreateRenderer
+    }
+    
+    guard TessResultRendererBeginDocument(renderer, "Unkown Title") == 1 else {
+      TessDeleteResultRenderer(renderer)
+      throw SwiftyTesseractError.unableToBeginDocument
+    }
+    
+    var pixImages = [Pix]()
+    defer {
+      // Release the Pix instance from memory
+      for var pix in pixImages {
+        pixDestroy(&pix)
+      }
+      semaphore.signal()
+    }
+    
+    for page in 0..<images.count {
+      
+      // convert image
+      let pixImage = try createPix(from: images[page])
+      pixImages.append(pixImage)
+  
+      // add image to document
+      guard TessBaseAPIProcessPage(tesseract, pixImage, Int32(page), "page.\(page)", nil, 0, renderer) == 1 else {
+        TessDeleteResultRenderer(renderer)
+        throw SwiftyTesseractError.unableToProcessPage
+      }
+    }
+    
+    guard TessResultRendererEndDocument(renderer) == 1 else {
+      TessDeleteResultRenderer(renderer)
+      throw SwiftyTesseractError.unableToEndDocument
+    }
+    
+    // process finished => delete the created renderer
+    TessDeleteResultRenderer(renderer)
```

Comment:
`TessDeleteResultRenderer` can just be called once in a defer block. I'd suggest adding this defer block after you assign a renderer (like after line 205):
```swift
defer {
    TessDeleteResultRenderer(renderer)
}
```
Defer statements are executed whenever the function is exited, including when errors are thrown

Reply:
I thought about that, but it is not possible because `TessDeleteResultRenderer(renderer)` needs to be called before this line:
https://github.com/SwiftyTesseract/SwiftyTesseract/blob/e486ccdbba71a99b5f90a191a3bf28b9a28232b9/SwiftyTesseract/SwiftyTesseract/SwiftyTesseract.swift#L245

We can wrap the whole logic in a new function and add a `defer` statement. This might be even cleaner. What do you think?